--- a/setup.py
+++ b/setup.py
@@ -115,6 +115,9 @@
     if bare_metal_version >= Version("12.8"):
         cc_flag.append("-gencode")
         cc_flag.append("arch=compute_100,code=sm_100")
+    if bare_metal_version >= Version("12.8"):
+        cc_flag.append("-gencode")
+        cc_flag.append("arch=compute_120,code=sm_120")
 
 
     # HACK: The compiler flag -D_GLIBCXX_USE_CXX11_ABI is set to be the same as
